pgmpy/factors/Factor.py

```diff
@@ -127,6 +127,9 @@ def get_cardinality(self, variables):
         >>> phi.get_cardinality(['x1', 'x2'])
         {'x1': 2, 'x2': 3}
         """
+        if type(variables)==str:
+            raise TypeError("List or array-like parameter expected, got string")
```

Comment:
@yashu-seth: Make the error message a bit more informative. Like you can include the argument name which you are expecting to be a list. Rather than showing just `string` write `type str`.